typingsgen.js

```diff
@@ -0,0 +1,74 @@
+var fs = require('fs');
+var regex = /export interface .*?Operators<T> \{([\S|\s]*)\}/;
+
+var core = fs.readFileSync('./src/CoreOperators.ts').toString();
+var kitchenSink = fs.readFileSync('./src/Rx.KitchenSink.ts').toString();
+var combinedMethods = core.match(regex)[1].trim() + '\n' + kitchenSink.match(regex)[1].trim();
+var contents = combinedMethods.split('\n');
+
+var operators = {};
+var fileResult = '/* tslint:disable:class-name */ /* tslint:disable:no-unused-variable */ /* tslint:disable:max-line-length */\n\
+import {Observable} from \'./Observable\';\n\
+import {ConnectableObservable} from \'./observable/ConnectableObservable\';\n\
+import {Scheduler} from \'./Scheduler\';\n\
+import {Notification} from \'./Notification\';\n\
+import {Subject} from \'./Subject\';\n\
+import {Observer} from \'./Observer\';\n\
+import {GroupedObservable} from \'./operator/groupBy-support\';\n\
+import {GroupByObservable} from \'./operator/groupBy\';\n\
+import {TimeInterval} from \'./operator/extended/timeInterval\';\n\
+import {ObservableOrPromise, ArrayOrIterator, _Selector, _IndexSelector, _SwitchMapResultSelector, _ObservableMergeMapProjector, _IteratorMergeMapProjector, _Predicate, _PredicateObservable, _Comparer, _Accumulator, _MergeAccumulator} from \'./types\';\n\n\
+';
+
+for (var i = 0; i < contents.length; i++) {
+  var item = contents[i].trim();
+  if (item) {
+    var file = item.match(/(.*?)\: operator.operator_proto_(.*?)<T>;/);
+    if (!file) {
+      continue;
+    }
+
+    var filename = file[2].trim();
+    var fileContent;
+
+    if (fs.existsSync('./src/operator/' + filename + '.ts')) {
+        fileContent = fs.readFileSync('./src/operator/' + filename + '.ts').toString();
+    } else {
+        fileContent = fs.readFileSync('./src/operator/extended/' + filename + '.ts').toString();
+    }
+
+    var methods = [];
+
+    var r = new RegExp('export function [_]?' + filename + '([\\s|\\S]*?[\\;\\{])', 'g');
+
+    do {
+      var result = r.exec(fileContent);
+      if (result) {
+        var method = result[1].trim();
+        if (methods.length > 0 && method.indexOf('{') > -1) {
+          continue;
+        }
+
+        method = method.split(/\n/g)
+          .filter(function (x) { return !!x; })
+          .map(function (x) { return ('' + x).trim(); })
+          .join(' ')
+          .replace(/ = .*?([\,|\)])/g, '$1');
+
+        if (method[method.length - 1] === ';' || method[method.length - 1] === '{') {
+          method = method.substr(0, method.length - 1).trim();
+        }
+
+        method = method.replace(/^<T>/, '').replace(/^<T, /, '<');
+        methods.push(method);
+      }
+    } while(result);
+
+    if (!operators[filename]) {
+      operators[filename] = true;
+      fileResult += 'export interface operator_proto_' + filename + '<T> {\n  ' + methods.join(';\n  ') + ';\n}\n';
+    }
+  }
+}
+
+fs.writeFileSync('./src/operator-typings.ts', fileResult);
```

Comment:
I see the above stated "crazy number of definitions" was generated by this script. Pretty cool, but again, is this something that's really necessary? How does it help? Do we have an example?